pkg/provisioner/ironic/ironic.go

```diff
@@ -1295,3 +1326,83 @@ func (p *ironicProvisioner) PowerOff() (result provisioner.Result, err error) {
 
 	return result, nil
 }
+
+// Retrieve configuration in form of Ironic's clean steps
+func (p *ironicProvisioner) buildCleanSteps(ironicNode *nodes.Node) (cleanSteps []nodes.CleanStep) {
+	// Check whether the clean steps for RAID configuration should be executed or not
+	if len(ironicNode.RAIDConfig) == 0 && p.host.Spec.RAID != nil {
+		raidCleanSteps := p.buildRAIDCleanSteps(ironicNode)
+		cleanSteps = append(cleanSteps, raidCleanSteps...)
+	}
+	return
+}
+
+// A private method to build the clean steps for RAID configuration from BaremetalHost spec
+func (p *ironicProvisioner) buildRAIDCleanSteps(ironicNode *nodes.Node) (cleanSteps []nodes.CleanStep) {
+	// ‘create_configuration’ doesn’t remove existing disks. It is recommended
+	// to add ‘delete_configuration’ before ‘create_configuration’ to make sure
+	// that only the desired logical disks exist in the system after manual cleaning.
+	cleanSteps = []nodes.CleanStep {{Interface: "raid", Step: "delete_configuration"}}
+
+	logialDisks := p.buildTargetRAIDCfg(p.host.Spec.RAID)
+	if logialDisks != nil {
+		err := nodes.SetRAIDConfig(
```

Comment:
This is a call to ironic, right? I don't think we want that as a side-effect of a method that claims to build a set of other instructions. We should do this right before we call ironic to provision an image.

Reply:
Done. I moved this method in to `startManualCleaning`